Basic/Invaders/Assets/Scripts/EnemyAgent.cs

```diff
@@ -4,22 +4,63 @@
 using UnityEngine.Assertions;
 using Random = UnityEngine.Random;
 
-public class AlienInvader : NetworkBehaviour
+public class EnemyAgent : NetworkBehaviour
 {
     private const float k_ShootingRandomThreshold = 0.996f;
-    private const float k_ShootTimer = 0.75f;
-    [Header("Alien Settings")]
+    private const float k_ShootTimer = 1.25f;
+    [Header("Enemy Settings")]
     public int score = 50;
     public GameObject bulletPrefab;
+    public float GraceShootingPeriod = 1.0f; // A period of time in which the enemy will not shoot at the start
 
     public bool canShoot { get; set; }
     public float column { get; private set; }
     public float row { get; private set; }
 
     private float m_ShootTimer = 0.0f;
+    private float m_FirstShootTimeAfterSpawn = 0.0f;
 
+    public void Awake()
+    {
+        canShoot = false;
+        m_FirstShootTimeAfterSpawn = Single.PositiveInfinity;
+    }
+
+    public override void NetworkStart()
+    {
+        base.NetworkStart();
+
+        if (IsServer)
+        {
+            canShoot = false;
+            if (score == 100)
+                return;
+
+            m_FirstShootTimeAfterSpawn =
+                Time.time + Random.Range(GraceShootingPeriod - 0.1f, GraceShootingPeriod + 0.75f);
+
+            Assert.IsTrue(InvadersGame.Singleton);
```

Comment:
```suggestion
            Assert.IsNotNull(InvadersGame.Singleton);
```
I suggest to not use the implicit boolean conversion of MonoBehaviour because it's confusing to new people.